md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/AppDeploymentQueueAPI.java

```diff
@@ -46,9 +45,201 @@
+            for (com.wipro.ats.bdre.md.dao.jpa.Process process1 : daoProcessList){
+                List<com.wipro.ats.bdre.md.dao.jpa.Properties> daoPropertiesList = propertiesDAO.getByProcessId(process1);
+                for (com.wipro.ats.bdre.md.dao.jpa.Properties daoProperties : daoPropertiesList) {
+                    Properties tableProperties = new Properties();
+                    tableProperties.setProcessId(daoProperties.getProcess().getProcessId());
+                    tableProperties.setConfigGroup(daoProperties.getConfigGroup());
+                    tableProperties.setKey(daoProperties.getId().getPropKey());
+                    tableProperties.setValue(daoProperties.getPropValue());
+                    tableProperties.setDescription(daoProperties.getDescription());
+                    propertiesList.add(tableProperties);
+                }}
+            processExport.setProcessList(processList);
+            processExport.setPropertiesList(propertiesList);
+            LOGGER.info("export object is "+processExport);
+            AddJson addJson=new AddJson();
+            String status=addJson.addJsonToProcessId(appDeploymentQueue.getProcessId().toString(),processExport);
+            LOGGER.info("status of process.json addition "+status);
+            com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue jpaAppDeploymentQueue=new com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue();
+            jpaAppDeploymentQueue.setAppDeploymentQueueStatus(appDeploymentQueueStatusDAO.get((short) 0));
+            jpaAppDeploymentQueue.setProcess(processDAO.get(appDeploymentQueue.getProcessId()));
+            jpaAppDeploymentQueue.setAppDomain(appDeploymentQueue.getAppDomain());
+            jpaAppDeploymentQueue.setAppName(appDeploymentQueue.getAppName());
+            jpaAppDeploymentQueue.setUsers(usersDAO.get(principal.getName()));
+            Long adqId=appDeploymentQueueDAO.insert(jpaAppDeploymentQueue);
+            LOGGER.info("app deployment queue Id is "+adqId);
+             returnedAppDeploymentQueue.setProcessId(appDeploymentQueue.getProcessId());
+             returnedAppDeploymentQueue.setAppName(appDeploymentQueue.getAppName());
+             returnedAppDeploymentQueue.setAppDomain(appDeploymentQueue.getAppDomain());
+             returnedAppDeploymentQueue.setUsername(principal.getName());
+             returnedAppDeploymentQueue.setAppDeploymentQueueId(adqId);
+             returnedAppDeploymentQueue.setAppDeploymentStatusId((short) 0);
+             restWrapper = new RestWrapper(returnedAppDeploymentQueue, RestWrapper.OK);
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+
+    @RequestMapping(value = "/merge/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public
+    RestWrapper merge(@PathVariable("id") Long queueId, Principal principal) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "merge" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)